Remotion/ObjectBinding/Web.ClientScript/Scripts/BocReferenceValueBase.ts

```diff
@@ -0,0 +1,223 @@
+
+    var newCommand = $('<a/>');
+
+    var tempCommandInfo: CommandInfo = {
+      href: null,
+      target: null,
+      onClick: null,
+      title: null
+    }
+    if (commandInfo != null)
+    {
+      tempCommandInfo = jQuery.extend(true, {}, commandInfo);
+      if (businessObject == null)
+      {
+        tempCommandInfo.href = null;
+        tempCommandInfo.target = null;
+        tempCommandInfo.onClick = null;
+        tempCommandInfo.title = null;
+      }
+      else if (commandInfo.href != null)
+      {
+        tempCommandInfo.href = String.format(commandInfo.href, businessObject);
+      }
+    }
+
+    var oldCommandAttributes = oldCommand[0].attributes;
+
+    for (var i = 0; i < oldCommandAttributes.length; i++)
+    {
+      var value = oldCommandAttributes[i].nodeValue;
+      if (value != null && value != '')
+        newCommand.attr(oldCommandAttributes[i].nodeName, value);
+    }
+
+    for (var property in tempCommandInfo)
+    {
+      var value: Nullable<string> = "" + (tempCommandInfo as any)[property]; // TODO is there a better way to do this?
+      if (value == null)
+        newCommand.removeAttr(property);
+      else
+        newCommand.attr(property, value);
+    }
+
+    newCommand.removeClass('hasIcon');
+
+    return newCommand;
+  };
+
+  public static CreateEmptyIcon (oldIcon: JQuery, title: string): JQuery
+  {
+    ArgumentUtility.CheckNotNull('oldIcon', oldIcon);
+    ArgumentUtility.CheckTypeIsString('title', title);
+  
+    var newIcon = oldIcon.clone();
+    newIcon.attr({ src: BocReferenceValueBase._nullIconUrl, alt: '' });
+    newIcon.removeAttr('title');
+    if (!StringUtility.IsNullOrEmpty(title))
+      newIcon.attr({ title: title });
+    newIcon.css({ width: oldIcon.width(), height: oldIcon.height() });
+  
+    return newIcon;
+  };
+
+  public static UpdateIconFromWebService (command: JQuery, icon: JQuery, iconInformation: IconInformation): void
+  {
+    ArgumentUtility.CheckNotNull('icon', icon);
+    ArgumentUtility.CheckTypeIsObject('iconInformation', iconInformation);
+  
+    if (iconInformation == null)
+      return;
+  
+    icon.attr({ src: iconInformation.Url });
+  
+    icon.attr({ alt: '' });
+    if (!StringUtility.IsNullOrEmpty(iconInformation.AlternateText))
+      icon.attr({ alt: iconInformation.AlternateText });
+  
+    if (!StringUtility.IsNullOrEmpty(iconInformation.ToolTip) && StringUtility.IsNullOrEmpty(icon.prop('title')))
+      icon.attr({ title: iconInformation.ToolTip });
+  
+    icon.css({ width: iconInformation.Width, height: iconInformation.Height });
+  
+    command.addClass('hasIcon');
+  };
+
+  public static ResetCommand (command: JQuery): void
+  {
+    ArgumentUtility.CheckTypeIsObject('command', command);
+    command.removeAttr('href');
+    command.removeAttr('onclick');
+    command.removeAttr('title');
+    command.removeAttr('target');
+    command.removeClass('hasIcon');
+  };
+}
+
```

Comment:
?

Reply:
I'm afraid I don't know what `?` exactly means, please take a moment to appreciate this comment from my point of view

![image](https://user-images.githubusercontent.com/24362621/111598525-9902a700-87cf-11eb-9e02-b1daad368b54.png)

Reply:
There's a bunch ob empty lines starting right at the ? and going all the way down to the end of the file. I do admit, I thought it would enough of a hint to point right at the issue without spelling it out. My bad.

Reply:
I'm afraid spelling these things out is necessary probably more often than not, especially without the proper context. I would have understood if you had made a multi line comment showing most of the empty lines (those are there btw because it's very much a work in progress after all, I didn't review it myself yet because I'm working on the other files)

Nonetheless, thanks for clarifying that bit, now I get it

Reply:
Ah, multi-line comments, that's a thing? Cool. The comment overview instead of looking at the sources, there is often an embarassing lack of context. Github could do better in this type of situation.

Reply:
yes the multi line thing actually helps a lot

![hZntfql633](https://user-images.githubusercontent.com/24362621/111603267-74f59480-87d4-11eb-962b-8674cbca5d6f.gif)

Reply:
my cursor looks like it had too much coffee but you get the gist